build-steps/release/check-uploaded-artifacts.gradle

```diff
@@ -82,6 +83,18 @@ def checkNoThirdParty = { JarFile jarFile ->
     assert jarFile.getEntry('com/tngtech/archunit/thirdparty') == null: 'There exists a third party folder'
 }
 
+def checkNoIllegalJarEntries = { JarFile jarFile ->
```

Comment:
Ah, very nice! 👍
Could we hook that into the regular build? If I understand correctly, this would only fail _after_ a release has already been published, right?

Reply:
I thought about that, too, but then refrained from it, because I was wondering if we should make the regular build more complex by that. But I guess it's worth it, since it's way better to catch this error at merge time, instead of at release time :+1: 
So I've adjusted all the jar file checks to run on every build and they are quite fast AFAIS.
"After a release has been published" is not completely correct though, we would catch it at the staging release, i.e. before it is really published, but at the very last moment before we would actually publish it to Maven Central. But anyway, as I said, now it should be caught at the moment something goes wrong with the dependencies :wink: